### ISSUE TITLE
`this` type parameter only works if there are no other parameters

### ISSUE THREAD

**Issue by STRd6** [DanielXMoore/Civet#446]
```typescript
fn := (this: T, x: X) ->
```

Actual:
```typescript
const fn = { this: T, x: X }(function () {})
```

Expected:
```typescript
function fn({ this: T, x: X }) {}
```

**Comment by edemaine** [comment]
We should support the `@` shorthand too.

**Comment by STRd6** [comment]
Fixed in v0.5.86
